src/optional-listeners/message-command-listeners/CorePrefixedMessage.ts

```diff
@@ -19,20 +20,32 @@ export class CoreListener extends Listener<typeof Events.PrefixedMessage> {
 		const spaceIndex = prefixLess.indexOf(' ');
 		const commandName = spaceIndex === -1 ? prefixLess : prefixLess.slice(0, spaceIndex);
 		if (commandName.length === 0) {
-			client.emit(Events.UnknownCommandName, { message, prefix, commandPrefix });
+			client.emit(Events.UnknownMessageCommandName, { message, prefix, commandPrefix });
 			return;
 		}
 
 		// Retrieve the command and validate:
 		const command = stores.get('commands').get(client.options.caseInsensitiveCommands ? commandName.toLowerCase() : commandName);
 		if (!command) {
-			client.emit(Events.UnknownCommand, { message, prefix, commandName, commandPrefix });
+			client.emit(Events.UnknownMessageCommand, { message, prefix, commandName, commandPrefix });
+			return;
+		}
+
+		// If the command exists but is missing a message handler, emit a different event (maybe an application command variant exists)
+		if (!command.messageRun) {
+			client.emit(Events.CommandDoesNotHaveMessageCommandHandler, { message, prefix, commandPrefix, command });
 			return;
 		}
 
 		// Run the last stage before running the command:
+		// TODO(vladfrangu): substr is deprecated!
```

Comment:
Make an issue or replace with `.substring` in the PR

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#the_difference_between_substring_and_substr

Reply:
Unresolving this @vladfrangu. There is no GH issue nor is the `substr` replaced.